CONTRIBUTING.md

```diff
@@ -45,22 +45,119 @@ To add a new plugin you need to do the following.
-  - `maintainers`: Register yourself as a maintainer for this plugin:
-    - e.g. `maintainers = [ lib.maintainers.JosephFourier ]`
-    - If you are already registered as a [`nixpkgs` maintainer][nixpkgs-maintainers], there is nothing more to do.
-    - Otherwise, you should add yourself to our [`maintainers.nix`](lib/maintainers.nix) list.
-    - See the documentation at the top of the [`nixpkgs` maintainers list][nixpkgs-maintainers] for more detail.
-  - `settingsOptions`: All or some (only the most important ones) option declarations for this plugin settings.\
-    See below for more information
-  - `settingsExample`: An example of what could the `settings` attrs look like.
-
-[nixpkgs-maintainers]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix
+3. Add the necessary parameters for the [`mkNeovimPlugin`](#mkneovimplugin)/[`mkVimPlugin`](#mkvimplugin)
+
+#### `mkNeovimPlugin`
+
+The `mkNeovimPlugin` function provide a standardize way to create a `Neovim` plugin.
+
+| Parameter                  | Description                                                                                                                                                                                                 | Required | Default Value |
+|----------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------|---------------|
+| **name**                   | The name of the plugin.                                                                                                                                                                                      | Yes      | N/A           |
+| **maintainers**            | Maintainers for the plugin.                                                                                                                                                                                  | Yes      | N/A           |
+| **url**                    | The URL of the plugin's repository.                                                                                                                                                                          | Yes      | If not set, `package`'s' `homepage` attribute |
+| **package**                | The nixpkgs package attr for this plugin. Can be a string, a list of strings, a module option, or any derivation. For example, "foo-bar-nvim" for `pkgs.vimPlugins.foo-bar-nvim`, or `[ "hello" "world" ]` will be referenced as `pkgs.hello.world`. | No       | `name` parameter |
+| **imports**                | Additional Nix modules to import.                                                                                                                                                                            | No       | `[]`            |
```

Comment:
```suggestion
| **imports**                | Additional modules to import.                                                                                                                                                                            | No       | `[]`            |
```

I never know how to word it when referencing modules.

"Nix modules" is wrong, because the module system comes from nixpkgs/nixos _not_ from nix itself.

"Nixpkgs modules" or "NixOS modules" could be used, but could also be confusing.

"Nixvim modules" is also legitimate, because the modules are intended for use with nixvim.

Or just "modules" on its own is probably enough, and removes the need to think about the additional word :sweat_smile: